lang/csharp/src/apache/main/Reflect/Readme.md

```diff
@@ -0,0 +1,198 @@
+# Namespace Avro.Reflect
+
+This namespace contains classes that implement Avro serialization and deserialization for plain C# objects. The classes use .net reflection to implement the serializers. The interface is similar to the Generic and Specific serialiation classes.
+
+## Serialization
+
+The approach starts with the schema and interates both the schema and the dotnet object together in a depth first manner per the specification. Serialization is the same as the Generic serializer except where the serializer encounters:
+- *A fixed type*: if the corresponding dotnet object type is a byte[] of the correct length then the object is serialized, otherwise an exception is thrown.
+- *A record type*: the serializer matches the schema property name to the dotnet object property name and then reursively serializes the schema property and the dotnet object property
+- *An array type*: See array serialization/deserialization.
+
+Basic serialization is performed as in the following example:
+
+```csharp
+    Schema schema; // created previously
+    T myObject; // created previously
+
+
+    var avroWriter = new ReflectWriter<T>(schema);
+    using (var stream = new MemoryStream(256))
+    {
+        avroWriter.Write(logMessage, new BinaryEncoder(stream));
```

Comment:
I think you meant `myObject`, rather than `logMessage` on this line.

```suggestion
        avroWriter.Write(myObject, new BinaryEncoder(stream));
```